### PR TITLE
Refactor FXIOS-4995 [v114] Update BreachAlertsManager theming

### PR DESCRIPTION
**Pull request by PARAIPAN9**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-4995)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/12019)

### Description
Update BreachAlertsManager theming system

## Logins Screen 
| Dark  | Light | 
| ------------- | ------------- |
| darkPasswordsScreen | lightPasswordsScreen |

## Saved Login Details
| Dark  | Light | 
| ------------- | ------------- |
| dark | light |

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/LoginManagement/BreachAlertsDetailView.swift
Comment (nishant2718) [inline Client/Frontend/LoginManagement/BreachAlertsDetailView.swift:217]: I believe the team would like to move off using extensions where possible. Instead, I would add `ThemeApplicable` to line 8 and add `applyTheme` towards the bottom of the class.

**Comment by nishant2718** [comment]
I noticed we're using UIColor.white in `BreachAlertsDetailView`. This isn't in our color system, so maybe @cwzilla can give us the right color to use there. 

@cwzilla can you let us know what the correct color to use would be for the warning text showing in Saved Login Details screenshots? Right now it's white in both light and dark. Do you want us to continue using white? Currently for primary text in light, we use `DarkGrey90` and in Dark we use `LightGrey05`.

File: Client/Frontend/LoginManagement/BreachAlertsDetailView.swift
Comment (nishant2718) [reply Client/Frontend/LoginManagement/BreachAlertsDetailView.swift:217]: We have a wiki page up for the view on extensions as well, if you'd like to read. https://github.com/mozilla-mobile/firefox-ios/wiki/When-To-Use-Extensions

**Comment by cwzilla** [comment]
@nishant2718 are you referring to the large red breach card? I think we should actually make some additional changes to it because the dark mode version isn't accessible and if we change the text to a dark grey, it might clash with the red background. I mocked up an alternative style using different colours, let me know what you think! 

https://www.figma.com/file/HiZmmpoICBKc9d58FDf5zh/Breach-Card?node-id=1%3A528&t=kuzorTcMaxRJjrnE-1

I noticed one other small thing on the main Login screen - for Light mode, the switch doesn't look like it's using actionPrimary. here is the [Figma file](https://www.figma.com/file/YIbYarab5aYte7KbK0bcVw/iOS-Components?node-id=1230%3A1185&t=aPa2KKwhXoOf3So0-1) for reference.

File: Client/Frontend/LoginManagement/BreachAlertsManager.swift
Comment (nishant2718) [inline Client/Frontend/LoginManagement/BreachAlertsManager.swift:28]: Since we're working in this area, I think it'd be good to clean up a little bit more. 

We can remove this static declaration and just have `UIImage(named: "Breached Website")?.withRenderingMode(.alwaysTemplate)` where it's needed. We can also add `Breached Website` into ImageIdentifiers.

**Comment by PARAIPAN9** [comment]
@cwzilla I updated the large breach cards above, using the colors that you provided. In light mode with layer2 colors the card doesn't look good because is the same color as the background.

**Comment by cwzilla** [comment]
@PARAIPAN9 What color is the background? Do you think applying a shadow to the card would help?

**Comment by nishant2718** [comment]
The changes are looking good! After @cwzilla's suggestions land, I'll do another review.

**Comment by PARAIPAN9** [comment]
@cwzilla, I updated the View applying a shadow, I also updated the screenshots. From my side, now it looks good.

**Comment by cwzilla** [comment]
@PARAIPAN9 Shadow looks great, thank you!

Will the light mode switch color be updated as well?

**Comment by PARAIPAN9** [comment]
Yes, but I think it is better to open another issue for that matter, because sometimes it changes the right color, sometimes not, so it seems like a bug.

**Comment by cwzilla** [comment]
@PARAIPAN9 Oh gotcha, sounds good. Thanks!

File: Client/Frontend/Settings/LoginDetailViewController.swift
Comment (nishant2718) [inline Client/Frontend/Settings/LoginDetailViewController.swift:82]: Do we need these defaults?

Comment (PARAIPAN9) [reply Client/Frontend/Settings/LoginDetailViewController.swift:82]: Ohh, sorry, forgot to remove them.

**Comment by nishant2718** [comment]
Okay looks good! We'll wait for a BR run and then land it.

**Review by nishant2718 — APPROVED** [review]

**Pull request merged**
